boost-maven/boost-maven-plugin/src/main/java/io/openliberty/boost/maven/plugin/AbstractMojo.java

```diff
@@ -88,31 +99,34 @@ public void execute() throws MojoExecutionException {
     }
     
     protected RuntimeI getRuntimeInstance() throws MojoExecutionException {
-        if(runtime == null) {
+        if (runtime == null) {
             BoostLogger logger = BoostLogger.getInstance();
-            if (dependencies.containsKey(AbstractBoosterConfig.RUNTIMES_GROUP_ID + ":tomee")) {
-                logger.info("Detected TomEE as target Boost runtime");
-                String tomeeInstallDir = projectBuildDir + "/apache-tomee/";
-                String tomeeConfigDir = tomeeInstallDir + "conf";
-                runtime = new TomeeRuntime(dependencies, getExecutionEnvironment(), tomeeInstallDir, tomeeConfigDir, getMavenDependencyPlugin());
-            } else if (dependencies.containsKey(AbstractBoosterConfig.RUNTIMES_GROUP_ID + ":openliberty")) {
-                logger.info("Detected Open Liberty as target Boost runtime");
-                String runtimeGroupId = "io.openliberty";
-                String runtimeArtifactId = "openliberty-runtime";
-                String runtimeVersion = "19.0.0.3";
-                runtime = new LibertyRuntime(dependencies, getExecutionEnvironment(), project, getLog(), repoSystem, repoSession, remoteRepos, getMavenDependencyPlugin(), runtimeGroupId, runtimeArtifactId, runtimeVersion);
-            } else if (dependencies.containsKey(AbstractBoosterConfig.RUNTIMES_GROUP_ID + ":wlp")) {
-                logger.info("Detected WebSphere Liberty as target Boost runtime");
-                String runtimeGroupId = "com.ibm.websphere.appserver.runtime";
-                String runtimeArtifactId = "wlp-javaee7";
-                String runtimeVersion = "19.0.0.3";
-                runtime = new LibertyRuntime(dependencies, getExecutionEnvironment(), project, getLog(), repoSystem, repoSession, remoteRepos, getMavenDependencyPlugin(), runtimeGroupId, runtimeArtifactId, runtimeVersion);
-            }
-            else {
-                throw new MojoExecutionException("No target Boost runtime was detected. Please add a runtime and restart the build.");
+            RuntimeParams params = new RuntimeParams(dependencies, getExecutionEnvironment(), project, getLog(), repoSystem, repoSession, remoteRepos, getMavenDependencyPlugin());
+            try {
+                List<URL> pathUrls = new ArrayList<URL>();
+                for(String compilePathElement : project.getCompileClasspathElements()) {
+                    pathUrls.add(new File(compilePathElement).toURI().toURL());
+                }
+
+                URL[] urlsForClassLoader = pathUrls.toArray(new URL[pathUrls.size()]);
+                ClassLoader compileClassLoader = new URLClassLoader(urlsForClassLoader, this.getClass().getClassLoader());
+                List<Class<?>> runtimes = ClassFinderUtil.findClassesImplementing(RuntimeI.class, "boost.runtimes", compileClassLoader, logger);
+
+                if (runtimes.size() == 1) {
+                    Object o = runtimes.get(0).getConstructor(params.getClass()).newInstance(params);
```

Comment:
Can we just pass in the `parms` object here rather than creating a new instance?

Reply:
Yup, that's what we're doing. We're getting a constructor that takes a `RuntimeParams` object and then calling `newInstance()` on that constructor.

Reply:
Ah my mistake!  I missed the ")" after getClass() and thought we were calling newInstance on that.